src/common/dtpthread.c

```diff
@@ -0,0 +1,50 @@
+/*
+    This file is part of darktable,
+    copyright (c) 2009--2014 johannes hanika.
+
+    darktable is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    darktable is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with darktable.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <sys/resource.h>
+
+#include "common/dtpthread.h"
+#include <sys/time.h>
+#include <sys/resource.h>
+
+int dt_pthread_create(pthread_t *thread, void *(*start_routine)(void*), void *arg)
+{
+  pthread_attr_t attr;
+  size_t stacksize;
+  int ret;
+  struct rlimit rlim;
+
+  pthread_attr_init(&attr);
+  pthread_attr_getstacksize(&attr, &stacksize);
+
+  getrlimit(RLIMIT_STACK, &rlim);
```

Comment:
I get yelling from the compiler when a non-void function's return value is not handled. This has been standard behaviour in gcc/g++ and the Solaris Studio compilers for a very long time. It's also standard practice to handle this rather than ignoring the return. 

http://pubs.opengroup.org/onlinepubs/009695399/functions/getrlimit.html  doesn't specify a value for the max, but if you check the description here:
http://docs.oracle.com/cd/E36784_01/html/E36872/getrlimit-2.html#scrolltoc then you'll get quite a good description.